lib/Migration/Version3500Date20231115184458.php

```diff
@@ -56,6 +56,11 @@ public function changeSchema(IOutput $output, Closure $schemaClosure, array $opt
 			$mailboxesTable->addUniqueIndex(['account_id', 'name_hash'], $indexNew);
 		}
 
+		$nameHashColumn = $mailboxesTable->getColumn('name_hash');
+		if (!$nameHashColumn->getNotnull()) {
+			$nameHashColumn->setNotnull(true);
```

Comment:
Unsure about this one. 

The default is still NULL and the value is omitted on insert a not null violation is thrown. 
Do we need a new default as well?

```
-- auto-generated definition
create table oc_mail_mailboxes
(
    id                  serial
        primary key,
    name                varchar(255)                                 not null,
    account_id          integer                                      not null,
    attributes          varchar(255) default '[]'::character varying not null,
    delimiter           varchar(1)   default NULL::character varying,
    messages            integer                                      not null,
    unseen              integer                                      not null,
    selectable          boolean      default false,
    special_use         varchar(255) default '[]'::character varying not null,
    sync_new_lock       integer,
    sync_changed_lock   integer,
    sync_vanished_lock  integer,
    sync_new_token      varchar(255) default NULL::character varying,
    sync_changed_token  varchar(255) default NULL::character varying,
    sync_vanished_token varchar(255) default NULL::character varying,
    sync_in_background  boolean      default false,
    my_acls             varchar(32)  default NULL::character varying,
    shared              boolean      default false,
    name_hash           varchar(255) default NULL::character varying not null
);
```

Reply:
Don't we want the opposite, and make the column nullable?

*not null* and violation when inserting with out a value is expected

Reply:
When we change not from not null to null here, all rows should have their name_hash filled. And for inserts we have to make sure to always write with the hash. If we insert without, there will be an error. I think that is fine and nothing we can improve.